thunder/core/proxies.py

```diff
@@ -137,8 +137,19 @@ def replace(self, **changes):
         kwargs.update(changes)
         return Proxy(**kwargs)
 
-    def replace_name(self, name: str | None = None):
+    def replace_name(self, name: str | None = None, *, disambiguate=False):
         """Return a copy of this proxy with the given name."""
+        if disambiguate:
+            trc = get_tracectx()
+        else:
+            trc = None
+
+        if trc is not None:
+            name_prefix = name
+            cnt = 0
+            while trc.has_name(name):
+                name = f"{name_prefix}_{cnt}"
```

Comment:
wouldn't we need to increment `cnt`?

Reply:
Great catch. :) Apparently we only needed one level up to now...